### ISSUE TITLE
When editing a section, the newline before the section disappears

### ISSUE THREAD

**Issue by snan** [splitbrain/dokuwiki#3828]
Since updating from Hogfather to this new one, when we edit a section and save the newline that just preceded the section is gone. This is with the markdown plugin and with a custom template. Not sure how to troubleshoot/fix.

**Comment by labrat1984** [comment]
I am experiencing the same problem. It is VERY annoying. I hope the fix will be prioritized.

(By the way, I LOVE dokuwiki and have been using it for years.)

**Comment by villepreux** [comment]
Any news on this one? Does anyone has fix? It's a showstopper for updating dokuwiki when using markdown as the wiki syntax

**Comment by xrat** [comment]
I fail to reproduce this on my DokuWikis running Igor. I tried it on 3 instances: One without any Markdown plugin, one with https://github.com/clockoon/dokuwiki-plugin-commonmark/ and one with https://github.com/mizunashi-mana/dokuwiki-plugin-mdpage/ . In none of them preceding whitespace was removed after editing a section.

Also, if this is related to a plugin, please, report to the plugin authors.

Note that section editing with mdpage is known to be broken: https://github.com/mizunashi-mana/dokuwiki-plugin-mdpage/issues/69 I used the workaround from the 2nd comment there.

**Comment by snan** [comment]
It still happens occasionally

**Comment by snan** [comment]
One workaround is to insert an extra newline before saving

**Comment by villepreux** [comment]
> One workaround is to insert an extra newline before saving

Yes. Since we needed a quick fix, we added that newline in inc/Ui/Editor.php, in addTextarea

`$data['form']->addTextarea('wikitext','')->attrs($attr)->val("\n".$TEXT)`

That's a hotfix. Don't know yet if we introduced side effects or not, but at least we can edit sections again.
